Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Dockerfile
+
+# Use an official Ruby image with the desired Ruby version
+FROM ruby:3.1
+
+# Set build arguments for user ID and group ID
+ARG USER_ID
+ARG GROUP_ID
+
+# Create a group and user matching the host USER_ID and GROUP_ID to avoid permission issues
+RUN groupadd -g $GROUP_ID shopifygroup && \
+    useradd -l -u $USER_ID -g shopifygroup -m shopifyuser
+
+# Install dependencies: curl, git, Node.js, npm
+RUN apt-get update && apt-get install -y \
+    curl git build-essential
+
+# Set up Node.js version 18 using NodeSource repository
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get install -y nodejs
+
+
+# Install Shopify CLI globally using npm
+RUN npm install -g @shopify/cli@latest
```

Comment:
can this be installed per project instead? the application directory will be mounted and node_modules will be present. if volumes are mounted with permissions then we can workaround having to hammer permissions

Reply:
Thank you for the suggestion! I used Adam's Stencil implementation (https://github.com/aligent/stencil-docker) as a reference. I was thinking that installing the Shopify CLI globally in the Docker container would help create a consistent, isolated environment. This approach can prevent developers from encountering installation issues on their local machines, which can vary widely in configuration. By ensuring that everyone works with the exact same setup, we can reduce the need for troubleshooting and maintain consistency across the team.

Reply:
counter intuitively i feel like depending on which version of the build image (if images are always tagged with `latest`) they _may_ have different shopfiy/cli because on `@latest`.

installing per project will be more consistent because it'll be locked by a package lock file like `package-lock.json`